### PR TITLE
Ad refactor

### PR DESCRIPTION
**Pull request by tacaswell**

### CODE REVIEW

File: ophyd/controls/area_detector.py [hunk 1]
```diff
@@ -394,28 +368,64 @@ def describe(self):
...
+class AreaDetectorFSBulkEntry(AreaDetectorFileStore):
+    def deconfigure(self, *args, **kwargs):
+
+        for uid, i in self._uid_cache:
+            fs.insert_datum(self._filestore_res, str(uid), {'point_number': i})
+
+        super(AreaDetectorFileStore, self).deconfigure(*args, **kwargs)
+
+
+class AreaDetectorFSInterativeWrite(AreaDetectorFileStore):
```
Comment (ericdill) [inline ophyd/controls/area_detector.py]: I assume this is supposed to be "Interactive" not "Interative"

File: ophyd/controls/area_detector.py [hunk 2]
```diff
@@ -344,29 +340,7 @@ def configure(self, *args, **kwargs):
         super(AreaDetectorFileStore, self).configure(*args, **kwargs)
         self._uid_cache.clear()
         self._uid_cache_darkfield.clear()
-
-    def deconfigure(self, *args, **kwargs):
-
-        i = 0
-        m = 0
-
-        # It is late and i am tired and this should be done better ...
```
Comment (danielballan) [inline ophyd/controls/area_detector.py]: Tom has your back, @stuwilkins. Goodbye, C-style loop logic.

File: ophyd/controls/area_detector.py [hunk 3]
```diff
@@ -394,28 +368,64 @@ def describe(self):
...
+
+        for uid, i in self._uid_cache:
+            fs.insert_datum(self._filestore_res, str(uid), {'point_number': i})
+
+        super(AreaDetectorFileStore, self).deconfigure(*args, **kwargs)
+
+
+class AreaDetectorFSInterativeWrite(AreaDetectorFileStore):
+    def read(self):
+        val = super(AreaDetectorFileStore, self).read()
```
Comment (danielballan) [inline ophyd/controls/area_detector.py]: should be `super(AreaDetectorFSInterativeWrite)`

File: ophyd/controls/area_detector.py [hunk 4]
```diff
@@ -394,28 +368,64 @@ def describe(self):
...
 
 
-class AreaDetectorFileStoreHDF5(AreaDetectorFileStore):
+class AreaDetectorFSBulkEntry(AreaDetectorFileStore):
+    def deconfigure(self, *args, **kwargs):
+
+        for uid, i in self._uid_cache:
+            fs.insert_datum(self._filestore_res, str(uid), {'point_number': i})
+
+        super(AreaDetectorFileStore, self).deconfigure(*args, **kwargs)
```
Comment (danielballan) [inline ophyd/controls/area_detector.py]: should be `super(AreaDetectorFSBulkEntry)`

**Comment by danielballan** [comment]
Superceded by #98

**Pull request closed without merging**